### DISCUSSION TITLE
Usage of repeatable fields

### DISCUSSION THREAD

**Discussion by fpoort** [OP-TED/eForms-SDK#29] [Q&A] [answered]
I've got a couple of questions regarding the usage of the repeatable properties in the various json files.

1. The repeatable property occurs in both the fields.json and the noticesubtype json files. Is it alright to assume that the repeatable properties for a specific field are the same across all these files? 

2. What is the intended usage of having the repeatable propety in multiple files for the same field?

3. I came across a couple of fields which do not seem to add up. I've added an example below of one of the fields. Does the metadata in this case mean that we should only add a maximum of 1 RegulatoryDomain to PriorInformationNotice or this an actual inconsistency?

**Snippet of UBL-PriorInformationNotice-2.3.xsd**
```xml
<xsd:element ref="cbc:RegulatoryDomain" minOccurs="0" maxOccurs="unbounded"/>
```

**Snippet of fields.json**
```json
{
    "id" : "BT-01-notice",
    "parentNodeId" : "ND-0",
    "name" : "Procedure Legal Basis",
    "btId" : "BT-01",
    "xpathAbsolute" : "/*/cbc:RegulatoryDomain",
    "xpathRelative" : "cbc:RegulatoryDomain",
    "type" : "internal-code",
    "legalType" : "CODE",
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    }
...
}
```

**Comment by rousso** [comment]
Hi @fpoort ,

Indeed, the repeatability information "belongs" only to the field metadata in [fields.json](https://github.com/OP-TED/eForms-SDK/blob/0.6.2/fields/fields.json). We are aware of the issue and we are in the process of rectifying it. 

The JSON files provided under [notice-types](https://github.com/OP-TED/eForms-SDK/tree/0.6.2/notice-types) up until SDK 0.6.x, are just a copy of the notice type definitions used by eNotices2, See the related section of the [SDK documentation](https://docs.ted.europa.eu/eforms/0.6.0/notice-types/index.html).

The main goal of the upcoming eForms SDK 0.7.0 is to replace the notice type definitions with new versions that avoid several problems like the one you spotted. We are working to release it by July 8.

For the 3rd point in your question, I will ask a colleague to get back to you with a separate answer. Thank you for bringing it to our attention. 

Regards,
Yannis

**Comment by rousso** [comment, marked as answer]
Hi again @fpoort,

This is to follow-up on the third point in your question about the apparent discrepancy between the cardinality of  `cbc:RegulatoryDomain` (`BT-01-notice`) in `fields.json` and in the schema.

So here is the thing: `cbc:RegulatoryDomain` is a **UBL element**, that is being reused by eForms. UBL defines this as `maxOccurs="unbounded"` and that is why it appears like that in the schema. **However, eFroms** is not as permissive as  UBL, so `fields.json` restricts this field to one occurrence (not-repeatable).

I hope this covers you. In general, similar apparent "discrepancies" must exist for several other fields. For eForms, the correct definition is the one provided in `fields.json`. 

Regards,
Yannis
